python/ray/serve/tests/test_standalone2.py

```diff
@@ -1345,7 +1351,48 @@ async def f():
     # Unblock the first request.
     signal_actor.send.remote()
     assert ray.get(first_ref) == "hello"
+    serve.shutdown()
+
+
+@pytest.mark.parametrize(
+    "ray_instance",
+    [
+        {
+            "RAY_SERVE_REQUEST_PROCESSING_TIMEOUT_S": "0.1",
+            "RAY_SERVE_HTTP_MAX_REPLICA_FAILURE_RETRIES": "5",
+        },
+    ],
+    indirect=True,
+)
+def test_http_request_number_of_retries(ray_instance):
```

Comment:
can we also add a test where the replica crashes?

you can do this by having a signal or counter actor in the replica, if it isn't set then `os.exit` to forcefully crash

Reply:
done!